src/rv/ui/screens/LiveGameScreen.java

```diff
@@ -103,17 +103,27 @@ public void keyPressed(KeyEvent e) {
             }
             break;
         case KeyEvent.VK_L:
-            resetTimeIfExpired();
-            viewer.getNetManager().getServer().freeKick(true);
+            if (e.isShiftDown()) {
+                resetTimeIfExpired();
```

Comment:
The `resetTimeIfExpired()` calls can be outside the `if` / `else` to avoid duplication.